attestation-service/docs/policy.md

```diff
@@ -19,7 +19,23 @@ These dimensions can be used to return a detailed, but generic description of th
 More information about these trust claims, including what the numerical values of the claims represent,
 can be found [here](https://datatracker.ietf.org/doc/draft-ietf-rats-ar4si/).
 
-AS policies will be provide `input`.
+A policy can also optionally define a claim called `extensions`.
+Any extensions that are specified via this claim will be added to the attestation token.
+The extensions claim should contain an array of extension objects.
+An extension object should look like this:
+```json
+{
+    "name": "<name-of-extension">,
+    "index": "<index-of-extension">,
+    "value": ...
+}
+```
```

Comment:
There are two design patterns here. One is to allow generic extensions to be added to the AS policy, which requires the user to know the index and name. The advantage is strong extensibility, but the disadvantage is that it's cumbersome for users to write the policy and they also need to know which extensions are available.

The other pattern is to register an extension, named something like ext.confidential-containers-claims or ext.trustee-claims, and the user only needs to enter any value. The code of AS hardcodes the name and index. The advantage is that it's convenient for users to write, but the disadvantage is that it limits the scope of the extension.

Both makes sense and probably both can survive together.

Reply:
Yeah adding some built-in extensions could make sense in the future, but let's start with this and see what people use.